### PR TITLE
Add exampel of web, api, identity server collaboration

### PR DESCRIPTION
**Pull request by Pavel-Lapatin**
#### CHANGE TYPE

 - [ ] Major
 - [x] Minor
 - [ ] Patch

#### GITHUB ISSUE LINK

https://github.com/GodelTech/GodelTech.Microservices.Core/issues/38
#### IMPACT, DEPENDENCIES AND TEST CONSIDERATIONS

New example of web client, identity server  and api server collaboration is added. 
Integration and unit tests for all apps are added.

### CODE REVIEW

**Comment by DiamondDragon** [comment]
@Pavel-Lapatin Few general comments:
* Please make project names much sorter. Here are suggested examples: **Microservice.Api**, **Microservice.Identity**, **Microservice.UI**. At the moment long project names makes project impossible to build. I had to put them into folder named `a`. This is not trivial activity and could cause issues with non-prepared developer.
* Please remove all included tests. It's difficult to understand relation between projects when multiple projects included. In fact we only need three project to illustrate usage of our library.

File: REST API and UI Collaboration/README.md [hunk 1]
```diff
@@ -1,7 +1,73 @@
...
+There are two main use cases in this application to show authentication and authorization processes.
+
+Use case 1:
+Given authenticated user and authorized client
+When ask for protected api resources
+Then protected resources is available
+
+Steps to reproduce use case:
+
+- open web client home page
```
Comment (DiamondDragon) [inline REST API and UI Collaboration/README.md:28]: Link to home page would be useful here. I guess something like http://localhost:XXX

File: REST API and UI Collaboration/README.md [hunk 2]
```diff
@@ -1,7 +1,73 @@
...
+
+Expected use case for this application is the web application that interacts with protected resources on behalf of authenticated user.
+
+## GodelTech.Microservices.WebAndApiCollaboration.Web
+
+This is the client which performs Api calls on behalf of users.
+
+There are two main use cases in this application to show authentication and authorization processes.
+
+Use case 1:
```
Comment (DiamondDragon) [inline REST API and UI Collaboration/README.md:21]: I would highlight this line using header of third level (e.g. ###)

File: REST API and UI Collaboration/README.md [hunk 3]
```diff
@@ -1,7 +1,73 @@
...
+- open web client home page
+- if you are not authenticated click on "Login" button in the upper left corner
+- if you have already authenticated click on "Logout" button in the upper left corner and repeat step 2
+- you will be redirected into identity server for authentication process
+- click "Login" button (there are no credentials just to simplify example)
+- you will be redirected back to home page
+- click on "Protected Resources" button
+- you should see "Protected resource" on the page
+
+Use case 2:
```
Comment (DiamondDragon) [inline REST API and UI Collaboration/README.md:37]: I would highlight this header too.

File: REST API and UI Collaboration/src/GodelTech.Microservices.WebAndApiCollaboration.IdentityServer/GodelTech.Microservices.WebAndApiCollaboration.IdentityServer.csproj [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  <ItemGroup>
+    <PackageReference Include="GodelTech.Microservices.Core" Version="1.3.0" />
+    <PackageReference Include="IdentityModel" Version="4.4.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Reference Include="IdentityServer4">
+      <HintPath>C:\Users\p.lapatsin\Downloads\IdentityServer4-main\IdentityServer4-main\src\IdentityServer4\src\obj\Debug\netcoreapp3.1\IdentityServer4.dll</HintPath>
```
Comment (DiamondDragon) [inline REST API and UI Collaboration/src/GodelTech.Microservices.WebAndApiCollaboration.IdentityServer/GodelTech.Microservices.WebAndApiCollaboration.IdentityServer.csproj:16]: These lines make build to fail on my computer. Project file can be fixed as follows:
```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <OutputType>Exe</OutputType>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="GodelTech.Microservices.Core" Version="1.3.0" />
    <PackageReference Include="IdentityModel" Version="4.4.0" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
    <PackageReference Include="IdentityServer4" Version="4.0.4" />
  </ItemGroup>

</Project>

```

File: REST API and UI Collaboration/README.md (hunk 1 above)
Comment (Pavel-Lapatin) [reply REST API and UI Collaboration/README.md]: Link was added

File: REST API and UI Collaboration/README.md (hunk 2 above)
Comment (Pavel-Lapatin) [reply REST API and UI Collaboration/README.md]: Use case 1 was highlighted

File: REST API and UI Collaboration/README.md (hunk 3 above)
Comment (Pavel-Lapatin) [reply REST API and UI Collaboration/README.md]: Use case 2 was highlighted

File: REST API and UI Collaboration/src/GodelTech.Microservices.WebAndApiCollaboration.IdentityServer/GodelTech.Microservices.WebAndApiCollaboration.IdentityServer.csproj (hunk 4 above)
Comment (Pavel-Lapatin) [reply REST API and UI Collaboration/src/GodelTech.Microservices.WebAndApiCollaboration.IdentityServer/GodelTech.Microservices.WebAndApiCollaboration.IdentityServer.csproj]: Sorry for this. I investigated how Identity server works and forgot revert this. It is fixed now.

**Pull request merged**